crates/context_aware_config/src/api/dimension/utils.rs

```diff
@@ -28,3 +32,41 @@ pub fn get_all_dimension_schema_map(
 
     Ok(dimension_schema_map)
 }
+
+pub fn get_dimension_usage_context_ids(
+    key: &str,
+    conn: &mut PooledConnection<ConnectionManager<PgConnection>>,
+) -> superposition::Result<Vec<String>> {
+    let result: Vec<Context> = contexts.load(conn).map_err(|err| {
+        log::error!("failed to fetch contexts with error: {}", err);
+        db_error!(err)
+    })?;
+
+    let mut context_ids = vec![];
+    for context in result.iter() {
+        let condition = Cac::<Condition>::try_from_db(
+            context.value.as_object().unwrap_or(&Map::new()).clone(),
+        )
+        .map_err(|err| {
+            log::error!("generate_cac : failed to decode context from db {}", err);
+            unexpected_error!(err)
+        })?
+        .into_inner();
+
+        extract_dimensions(&condition)?
+            .get(key)
+            .map(|_| context_ids.push(context.id.to_owned()));
+    }
+    Ok(context_ids)
+}
+
+pub fn fetch_dimension(
+    key: &String,
+    conn: &mut PooledConnection<ConnectionManager<PgConnection>>,
+) -> superposition::Result<Dimension> {
+    let res = dimensions::dsl::dimensions
+        .filter(dimensions::dimension.eq(key))
+        .select(Dimension::as_select())
+        .get_result(conn)?;
+    Ok(res)
+}
```

Comment:
I think this is used in a single place, can we remove this function and just run the query directly in the handler?